components/veml7700/veml7700.h

```diff
@@ -0,0 +1,155 @@
+ *
+ * Copyright (c) 2022 Marc Luehr <marcluehr@gmail.com>
+ *
+ * ISC Licensed as described in the file LICENSE
+ */
+#ifndef __VEML7700_H__
+#define __VEML7700_H__
+
+#include <stddef.h>
+#include <i2cdev.h>
+#include <esp_err.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define VEML7700_I2C_ADDR (0x10)
+
+#define VEML7700_INTEGRATION_TIME_25MS  (0b1100)
+#define VEML7700_INTEGRATION_TIME_50MS  (0b1000)
+#define VEML7700_INTEGRATION_TIME_100MS (0b0000)
+#define VEML7700_INTEGRATION_TIME_200MS (0b0001)
+#define VEML7700_INTEGRATION_TIME_400MS (0b0010)
+#define VEML7700_INTEGRATION_TIME_800MS (0b0011)
+
+#define VEML7700_GAIN_1     (0b00)
+#define VEML7700_GAIN_2     (0b01)
+#define VEML7700_GAIN_DIV_8 (0b10)
+#define VEML7700_GAIN_DIV_4 (0b11)
+
+#define VEML7700_POWER_SAVING_MODE_500MS  (0b00)
+#define VEML7700_POWER_SAVING_MODE_1000MS (0b01)
+#define VEML7700_POWER_SAVING_MODE_2000MS (0b10)
+#define VEML7700_POWER_SAVING_MODE_4000MS (0b11)
+
+#define VEML7700_PERSISTENCE_PROTECTION_1 (0b00)
+#define VEML7700_PERSISTENCE_PROTECTION_2 (0b01)
+#define VEML7700_PERSISTENCE_PROTECTION_4 (0b10)
+#define VEML7700_PERSISTENCE_PROTECTION_8 (0b11)
+
+/**
+ * VEML configuration descriptor
+ */
+typedef struct
+{
+    uint16_t gain : 2;                //!< control the sensitivity
+    uint16_t integration_time : 4;    //!< time to measure
+    uint16_t persistence_protect : 2; //!< sample count before the interrupt triggers
+    uint16_t interrupt_enable : 1;    //!< enable threshold interrupt
+    uint16_t shutdown : 1;            //!< set to 1 to shutdown the device, set to 0 to wakeup
+    uint16_t threshold_high;          //!< high threshold for the interrupt
+    uint16_t threshold_low;           //!< low threshold for the interrupt
+    uint16_t power_saving_mode : 2;   //!< power saving mode
+    uint16_t power_saving_enable : 1; //!< enable the pover saving mode
+} veml7700_config_t;
+
+/**
+ * @brief Initialize device descriptor
+ *
+ * Default SCL frequency is 100kHz. The I2C address is fix.
+ *
+ * @param dev Pointer to I2C device descriptor
+ * @param port I2C port number
+ * @param sda_gpio SDA GPIO
+ * @param scl_gpio SCL GPIO
+ * @return `ESP_OK` on success
+ */
+esp_err_t veml7700_init_desc(i2c_dev_t *dev, i2c_port_t port, gpio_num_t sda_gpio, gpio_num_t scl_gpio);
+
+/**
+ * @brief Free device descriptor
+ *
+ * @param dev Pointer to I2C device descriptor
+ * @return `ESP_OK` on success
+ */
+esp_err_t veml7700_free_desc(i2c_dev_t *dev);
+
+/**
+ * @brief Probe if the device exist on the bus
+ *
+ * @param dev Pointer to I2C device descriptor
+ * @return `ESP_OK` on success
+ */
+esp_err_t veml7700_probe(i2c_dev_t *dev);
+
+/**
+ * @brief Write the config to the device
+ *
+ * @param dev Pointer to I2C device descriptor
+ * @param config Pointer to the config descriptor
+ * @return `ESP_OK` on success
+ */
+esp_err_t veml7700_config(i2c_dev_t *dev, veml7700_config_t *config);
```

Comment:
Function names must contain a verb. For example, in this case `veml7700_set_config`. In addition, there is no reverse function (`veml7700_get_config()`) - it should be possible to read the device configuration.

Reply:
Added a verb to all function names. I also added the reverse function and a check for it in the example.